samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java

```diff
@@ -253,9 +255,10 @@ private static void calculateIntStreamPartitions(JobGraph jobGraph, Config confi
     if (partitions < 0) {
       // use the following simple algo to figure out the partitions
       // partition = MAX(MAX(Input topic partitions), MAX(Output topic partitions))
+      // partition will be further bounded by MAX_INFERRED_PARTITIONS. This is important when running in hadoop.
```

Comment:
Please elaborate why it's important for hadoop. The current statement is to vague to be useful for code maintainers